server/controllers/stock/core.js

```diff
@@ -345,14 +347,18 @@ async function getMovements(depotUuid, params) {
     d.text AS depot_text, IF(is_exit = 1, "OUT", "IN") AS io,
     BUID(m.depot_uuid) AS depot_uuid, m.is_exit, m.date, BUID(m.document_uuid) AS document_uuid,
     m.flux_id, BUID(m.entity_uuid) AS entity_uuid, SUM(m.unit_cost * m.quantity) AS cost,
-    f.label AS flux_label, BUID(m.invoice_uuid) AS invoice_uuid, dm.text AS documentReference
+    f.label AS flux_label, BUID(m.invoice_uuid) AS invoice_uuid, dm.text AS documentReference,
+    BUID(sr.uuid) AS stock_requisition_uuid, sr_m.text AS document_requisition
   FROM stock_movement m
-    JOIN lot l ON l.uuid = m.lot_uuid
-    JOIN inventory i ON i.uuid = l.inventory_uuid
-    JOIN depot d ON d.uuid = m.depot_uuid
-    JOIN flux f ON f.id = m.flux_id
-    LEFT JOIN document_map dm ON dm.uuid = m.document_uuid
-    LEFT JOIN service AS serv ON serv.uuid = m.entity_uuid
+  JOIN lot l ON l.uuid = m.lot_uuid
+  JOIN inventory i ON i.uuid = l.inventory_uuid
+  JOIN depot d ON d.uuid = m.depot_uuid
+  JOIN flux f ON f.id = m.flux_id
+  LEFT JOIN document_map dm ON dm.uuid = m.document_uuid
+  LEFT JOIN service AS serv ON serv.uuid = m.entity_uuid
+  LEFT JOIN stock_requisition_movement smr ON smr.document_uuid = m.document_uuid
+  LEFT JOIN stock_requisition sr ON sr.uuid = smr.stock_requisition_uuid
+  LEFT JOIN document_map sr_m ON sr_m.uuid = sr.uuid
```

Comment:
You are adding a huge amount of JOINs to a query that is executed frequently!  Do we really need the requisition uuid with every movement, even if the depot rarely uses requisitions...?

I'm not really comfortable with this change.  Where is it that you need the stock requisition text?  Can you just look that up in a separate query?

Reply:
It was just because of this constraint where there was almost a link between the stock movements and the requirements.

![image](https://user-images.githubusercontent.com/8035588/94550735-25360780-024c-11eb-862a-960d8f321046.png)

Reply:
Yes, but this doesn't mean there needs to be a JOIN.  This just means there needs to be a link.

Reply:
I propose to add the requisition_uuid column on the stock_movement table in the same way as there is the invoice_uuid column
![image](https://user-images.githubusercontent.com/8035588/94658779-2a05c480-02fb-11eb-9220-32c15379e4a6.png)